### PR TITLE
Register and Update OAuth client via Stern/Backoffice

### PR DESCRIPTION
**Pull request by battermann**
## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: tools/stern/src/Stern/Intra.hs [hunk 1]
```diff
@@ -853,3 +858,64 @@ getUserNotifications uid = do
...
+    rpc'
+      "brig"
+      b
+      ( method GET
+          . Bilge.paths ["i", "oauth", "clients", toByteString' cid]
+      )
+  case statusCode r of
+    200 -> parseResponse (mkError status502 "bad-upstream") r
+    404 -> throwE (mkError status404 "bad-upstream" "not-found")
+    _ -> throwE (mkError status502 "bad-upstream" "")
```
Comment (fisx) [inline tools/stern/src/Stern/Intra.hs:891]:
```suggestion
    _ -> throwE (mkError status502 "bad-upstream" (show r))
```

(not tested; not that important, but might save some time if it breaks.)

File: tools/stern/test/integration/API.hs [hunk 2]
```diff
@@ -93,14 +95,34 @@ tests s =
       test s "GET /i/consent" testGetConsentLog,
       test s "GET /teams/:id" testGetTeamInfo,
       test s "GET i/user/meta-info?id=..." testGetUserMetaInfo,
-      test s "/teams/:tid/search-visibility" testSearchVisibility
+      test s "/teams/:tid/search-visibility" testSearchVisibility,
+      test s "i/oauth/clients" testCrudOAuthClient
```
Comment (fisx) [inline tools/stern/test/integration/API.hs:99]: why the `/i`?  stern doesn't have any public-facing api, so the distinction doesn't really work here i think.

File: tools/stern/test/integration/API.hs [hunk 3]
```diff
@@ -93,14 +95,34 @@ tests s =
...
       -- The following endpoints can not be tested because they require ibis:
       -- - `GET /teams/:tid/billing`
       -- - `GET /teams/:tid/invoice/:inr`
       -- - `PUT /teams/:tid/billing`
       -- - `POST /teams/:tid/billing`
     ]
 
+testCrudOAuthClient :: TestM ()
+testCrudOAuthClient = do
```
Comment (fisx) [inline tools/stern/test/integration/API.hs:108]: i won't insist, but maybe add new stern tests to `/integration` as well in the future?  it's also good to save time, though...

**Review by fisx — APPROVED** [review]

File: tools/stern/test/integration/API.hs (hunk 2 above)
Comment (battermann) [reply tools/stern/test/integration/API.hs:99]: I've seen it on other stern endpoints, too, so I adapted it. I guess, it just serves as a marker to make it obvious that this is calling an internal endpoint. Should I remove it?

File: tools/stern/test/integration/API.hs (hunk 3 above)
Comment (battermann) [reply tools/stern/test/integration/API.hs:108]: Yes, good point. Now the work is done. Let's keep it for now.

File: tools/stern/test/integration/API.hs (hunk 2 above)
Comment (fisx) [reply tools/stern/test/integration/API.hs:99]: when i complained about this i thought this was the only end-point in stern with an `/i`, and that stern is only calling internal end-points.  both assumptions may be wrong.  use your own judgement, if it's consistent with what stern does everywhere, keep it.

Comment (fisx) [reply tools/stern/test/integration/API.hs:99]: ah, i see i'm a bit late.  :)  either way it's not a big deal.

**Pull request merged**